### PR TITLE
F51 94 home view

### PR DESCRIPTION
**Pull request by amrarick26**
`Buyer.xp.Slides: {
    Interval: 5000,
    NoWrap: false,
    AutoPlay: true,
    Items: [
        {
            Src: "url",
            ID: "slideID",
            Title: "slideTitle",
            SubText: "slideSubText"
        }
    ]
}`

Interval: measured in milliseconds (if AutoPlay: false, Interval set to null)
NoWrap: disables the looping of slides. Setting no-wrap to an expression which evaluates to a truthy value will prevent looping.
AutoPlay: Automatically slides after set interval time

If no Items in Buyer.xp.Slides, carousel will use fall back images in assets/images directory.

### CODE REVIEW

File: src/app/common/directives/oc-carousel/oc-carousel.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        scope: {
+            buyer: '=',
+            slides: '='
+        },
+        restrict: 'E',
+        link: function(scope, element) {
+            var buyer = scope.buyer;
+            var slides = scope.slides;
+            if (buyer.xp && buyer.xp.Slides && buyer.xp.Slides.Items && buyer.xp.Slides.Items.length) {
+                console.log(buyer.xp.Slides.Items);
```
Comment (crhistianr) [inline src/app/common/directives/oc-carousel/oc-carousel.js]: remove console.log

File: src/app/home/js/home.config.js [hunk 2]
```diff
@@ -12,6 +12,11 @@ function HomeConfig($stateProvider) {
 			controllerAs: 'home',
 			data: {
 				pageTitle: 'Home'
+			},
+			resolve: {
+				Buyer: function(OrderCloudSDK, buyerid) {
+					return OrderCloudSDK.Buyers.Get(buyerid);
+				}
```
Comment (crhistianr) [inline src/app/home/js/home.config.js]: this is already a resolve on base, not necessary here

File: src/app/common/directives/oc-carousel/oc-carousel.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+angular.module('orderCloud')
+    .directive('ocCarousel', ocCarouselDirective)
+;
```
Comment (crhistianr) [inline src/app/common/directives/oc-carousel/oc-carousel.js]: We should have some error message display if there are no slides defined on the buyer and there are no slides being passed in to the  directive.

Checkout how I used `$exceptionHandler` to do this on ocReorder as an example

File: src/app/common/directives/oc-carousel/oc-carousel.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+function ocCarouselDirective($compile, $templateRequest) {
+    return {
+        scope: {
+            buyer: '=',
+            slides: '='
+        },
+        restrict: 'E',
+        link: function(scope, element) {
+            var buyer = scope.buyer;
+            var slides = scope.slides;
```
Comment (crhistianr) [inline src/app/common/directives/oc-carousel/oc-carousel.js]: slides is defined but never used

File: src/app/common/directives/oc-carousel/oc-carousel.js (hunk 3 above)
Comment (amrarick26) [inline src/app/common/directives/oc-carousel/oc-carousel.js]: Added

File: src/app/home/js/home.config.js (hunk 2 above)
Comment (amrarick26) [inline src/app/home/js/home.config.js]: Not in base.config

**Comment by spencerwalker** [comment]
replace code in `_carousel.less` with the following:
```
// Center images that do not fill the carousel element
.carousel-inner {
  > .item {
    img {
      margin: 0 auto;
    }
    > a {
      img {
        margin: 0 auto;
      }
    }
  }
}
```
This will center the images in the carousel if they do not fill the element and doesn't add anything on top of core bootstrap classes.

**Comment by amrarick26** [comment]
@crhistianr everything should be updated now

File: src/app/common/directives/oc-carousel/oc-carousel.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            slides: '='
+        },
+        restrict: 'E',
+        link: function(scope, element) {
+            if(typeof scope.buyer === 'undefined'){
+                $exceptionHandler({message: 'ocCarousel directive is not configured correctly, missing buyer object data'});
+            }
+            if((!scope.buyer) && typeof scope.slides === 'undefined'){
+                $exceptionHandler({message: 'ocCarousel directive is not configured correctly, missing slides data'});
+            }
```
Comment (crhistianr) [inline src/app/common/directives/oc-carousel/oc-carousel.js]: You're not handling the scenario where buyer is defined, but the slides property on buyer is not. In which case there would still not be anything displaying.

You could define a shorthand handler for slides and at the same time verify its existence. Maybe something like:
`var slides = (buyer.xp && buyer.xp.Slides && buyer.xp.Slides.Items && buyer.xp.Slides.Items.length) ? buyer.xp.Slides : null`

then use that for your $exceptionHandler and then below for your other conditional

**Comment by robertsoniv** [comment]
Buyer list/get are not normal shopper activities.  We need to wait until the currentUser has a buyer property; however, the directive setup should still work - it'll just pass through base.currentUser.Buyer instead.

**Comment by robertsoniv** [comment]
Also, I don't think the typical use case of a slider would be _inside_ of a jumbotron.  So the default images should be 1140px wide instead of 1020px.

**Pull request merged**